youtube_dl/extractor/tiktok.py

```diff
@@ -94,13 +103,74 @@ class TikTokIE(TikTokBaseIE):
     }]
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        webpage = self._download_webpage(
-            'https://m.tiktok.com/v/%s.html' % video_id, video_id)
-        data = self._parse_json(self._search_regex(
-            r'\bdata\s*=\s*({.+?})\s*;', webpage, 'data'), video_id)
-        return self._extract_aweme(data)
+        video_id = url.split('/')[-1]
+
+        # extract meta data using the official api
+        # Response json contains: provider url, title, html, author_namee, height, thumbnail_width, width, version,
+        # author_url, thumbnail_height, thumbnail_url, type, provider_name (tiktok)
+
+        json_api = self._download_json('https://www.tiktok.com/oembed?url=' + url, video_id)
+
+        # extract metadata with beautifulSoup
+        webpage = self._download_webpage(url, video_id)
+        soup = BeautifulSoup(webpage, features="html.parser")
+        h2 = soup.find_all("h2", {"class": "jsx-1038045583 jsx-3192540912 jsx-2150087249 video-meta-count"})
+        data = h2[0].text.split(' ')
+        likes_count = self.numeric_convert(data[0])
+        comments_count = self.numeric_convert(data[3])
+        json_next_data = soup.find(id='__NEXT_DATA__')
+        props = json_next_data.contents[0]
+        json_data_encode = json.dumps(props.encode('utf-8'))
+        ast_le = ast.literal_eval(json_data_encode)
+        data_dict = json.loads(ast_le)
+        timestamp = self.numeric_convert(data_dict['props']['pageProps']['videoData']['itemInfos']['createTime'])
+
+        shares = data_dict['props']['pageProps']['videoData']['itemInfos']['shareCount']
```

Comment:
I would add the line:
item_infos = data_dict['props']['pageProps']['videoData']['itemInfos'] 
to make these lines clearer